CountingBloomFilter.hpp

```diff
@@ -126,12 +117,11 @@ class CountingBloomFilter
 	size_t m_sizeInBytes;
 	unsigned m_hashNum;
 	unsigned m_kmerSize;
-	double m_dFPR;     // unused
-	uint64_t m_nEntry; // unused
-	uint64_t m_tEntry; // unused
 	static const uint32_t BloomFilter_VERSION = 2;
 	unsigned m_countThreshold;
 	unsigned m_bitsPerCounter = 8;
+	static constexpr const char* MAGIC_HEADER_STRING = "BLOOMCOU\0";
```

Comment:
```suggestion
	static constexpr const char* MAGIC_HEADER_STRING = "BLOOMCOU";
```
The "\0" is implicit, so you don't need it here. As it is you would actually get two terminating zeros, the explicit one and the implicit one. Alternatively you can do
```c++
static constexpr const char MAGIC_HEADER_STRING[] = "BLOOMCOU";
```
which is similar but slightly different. One big difference is that `sizeof(MAGIC_HEADER_STRING)` of the `char*` type returns the size of a pointer, 8 bytes, whereas `sizeof(MAGIC_HEADER_STRING)` of the `char[]` returns the size of the string including the terminating "\0" (I believe, good to double check). I have a slight preference for `[]`.

Reply:
I recommend that we use the same `MAGIC_HEADER_STRING` for both counting Bloom filters and non-counting Bloom filters, and in non-counting Bloom filters set `m_bitsPerCounter = 1`. That way the file format and serialization code can be shared between the two data structures. It avoids unnecessary code duplication and file format proliferation. I suggest we use a magic string of `BTLBloom` (eight characters).
cc @jowong4 @schutzekatze @JustinChu